### ISSUE TITLE
Remove deprecated t.python.util.unsignedID and setIDFunction

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6707]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#6707|
|Type|task|
|Created|2013-08-27 21:15:16Z|
|Branch|https://github.com/twisted/twisted/tree/unsigned-id-removal-6707|

`unsignedID` and `setIDFunction` in [twisted/python/util.py](twisted/python/util.py) have been deprecated since 13.0 and can be removed in an upcoming release.

Discovered as part of the Python 2.5 code review ticket ([#5715](https://github.com/twisted/twisted/issues/5715)).

Attachments:

* [6707.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/217/217a2089eed1fd0f407ea853a29608b1cbaf9bb2/216be5642517bf72f22edc25c966189e3ec28979.patch) (5563 bytes) - added by ewong on 2013-11-06 06:10:24Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__6707 6707
type__task task
reporter__thijs thijs
priority__low low
milestone__ 
branch__branches_unsigned_id_removal_6707 branches/unsigned-id-removal-6707
branch_author__ewong ewong
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1377638116000000 1377638116000000
changetime__1389476891000000 1389476891000000
version__None None
owner__exarkun exarkun
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_2"></a>|@tomprince commented|
|-|-|

Note that this can't be merged until at least 14.0, since it hasn't yet been a year since they were deprecated. 13.0 was released 2013-03-19.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [41028]) Branching to 'unsigned-id-removal-6707'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra set owner to @exarkun|
|-|-|

Thanks. There are 2 private vars left that only seem to be used by the removed functions:

 - `_idFunction`
 - `_HUGEINT`

Please remove those as well and merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [41206]) Merge unsigned-id-removal-6707

Author: ewong
Reviewer: thijs
Fixes: [#6707](https://github.com/twisted/twisted/issues/6707)

Remove `twisted.python.util.unsignedID` and `twisted.python.util.setIDFunction`,
deprecated since 13.0.
